File: BackEnd/src/main/java/tn/esprit/backend/entities/Event.java
package tn.esprit.backend.entities;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.Date;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Event {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    Long EventID;
    String EventName;
    String EventDescription;
    Date EventDate;
    Float Price;
    Long AvailablePlaces;
    Long EventPlaces;
    String Latitude;
    String Longitude;

    // Image data as byte array
    @Lob
    @Basic(fetch = FetchType.LAZY)
    byte[] imageData;

    @ManyToMany(cascade = CascadeType.ALL)
    Set<Hardware> hardwares;
}

File: FrontEnd/src/app/event.ts

import { Hardware } from './hardware';
export class Event {
    eventID: number; 
    eventName: string;
    eventDescription: string;
    eventDate: Date; 
    price: number; 
    availablePlaces: number; 
    eventPlaces: number;
    latitude: string; 
    longitude: string; 
    hardwares: Hardware[];
  }
  